duplocloud/resource_duplo_plan_settings.go

```
@@ -0,0 +1,185 @@
+						Optional: true,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourcePlanSettings() *schema.Resource {
+	return &schema.Resource{
+		Description: "`duplocloud_plan_settings` manages an plan settings in Duplo.",
+
+		ReadContext:   resourcePlanSettingsRead,
+		CreateContext: resourcePlanSettingsCreateOrUpdate,
+		UpdateContext: resourcePlanSettingsCreateOrUpdate,
+		DeleteContext: resourcePlanSettingsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(15 * time.Minute),
+		},
+		Schema: duploPlanSettingsSchema(),
+	}
+}
+
+func resourcePlanSettingsRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	planID := d.Id()
+	log.Printf("[TRACE] resourcePlanSettingsRead(%s): start", planID)
+
+	c := m.(*duplosdk.Client)
+
+	duplo, err := c.PlanGet(planID)
+	if err != nil && !err.PossibleMissingAPI() {
+		return diag.Errorf("failed to retrieve plan for '%s': %s", planID, err)
+	}
+
+	if duplo == nil {
+		plan, err := c.PlanGet(planID)
+		if err != nil {
+			return diag.Errorf("failed to read plan configs: %s", err)
+		}
+		if plan == nil {
+			return diag.Errorf("failed to read plan: %s", planID)
+		}
+
+		duplo = plan
+	}
+	flattenPlanSettings(d, duplo)
+	log.Printf("[TRACE] resourcePlanSettingsRead(%s): end", planID)
+	return nil
+}
+
+func resourcePlanSettingsCreateOrUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	planID := d.Get("plan_id").(string)
+	log.Printf("[TRACE] resourcePlanSettingsCreateOrUpdate(%s): start", planID)
+
+	c := m.(*duplosdk.Client)
+
+	duplo, err := c.PlanGet(planID)
+	if err != nil && !err.PossibleMissingAPI() {
+		return diag.Errorf("failed to retrieve plan for '%s': %s", planID, err)
+	}
+	if v, ok := d.GetOk("unrestricted_ext_lb"); ok {
```

We should check for a nil `duplo` result from the API call. If nil, the provider should return an error that the plan could not be found.

---

Done.